src/io/benchmark.zig

```diff
@@ -0,0 +1,238 @@
+const std = @import("std");
+const os = std.os;
+const assert = std.debug.assert;
+
+const Time = @import("time.zig").Time;
+const IO = @import("io.zig").IO;
+
+pub fn main() !void {
+    try struct {
```

Comment:
This `struct { fn run() void { ... } }.run();` pattern is nice for tests as it avoids polluting the outer scope. However I don't think it makes much sense for this benchmark code. Le'ts just do the standard
```zig
const Context = struct {};

pub fn main() void {
    // contents of the current run() function.
}
```